File: uniq-(UNIX)/uniq.js
const uniq = (array) => {
    if(array.length === 1) return [array[0]];

    return array.filter((element, index) => 
        element === null || element !== array[index + 1]
    );

};

module.exports = { uniq };



// return array.filter((element, index) => {
//     if(element === null) return element;
//     if(element !== array[index + 1]) return element;
// }); // filters the null values & fails test 4

//https://www.codewars.com/kata/52249faee9abb9cefa0001ee/train/javascript

File: python-venv/uniq/kata_function/uniq.py
def uniq(array):
    if len(array) == 1:
        return array

    out = []
    for i in range(0, len(array) - 1, 1):
        if i == 0:
            out.append(array[0])
        if array[i] != array[i + 1]:
            out.append(array[i + 1])

    return out

#https://www.codewars.com/kata/52249faee9abb9cefa0001ee/train/python